src/faction_camp.cpp

```diff
@@ -683,21 +683,36 @@ void basecamp::get_available_missions_by_dir( mission_data &mission_key, const p
                                     entry, avail );
         }
         // Generate upgrade missions for expansions
-        for( const basecamp_upgrade &upgrade : available_upgrades( dir ) ) {
-            const base_camps::miss_data &miss_info = base_camps::miss_info[ "_faction_upgrade_exp_" ];
-            comp_list npc_list = get_mission_workers( upgrade.bldg + "_faction_upgrade_exp_" +
+        std::vector<basecamp_upgrade> upgrades = available_upgrades( dir );
+        std::vector<size_t> order;
+
+        for( size_t i = 0; i < upgrades.size(); i++ ) {
+            order.push_back( i );
+        }
```

Comment:
`order` vector can be initialised like
```suggestion
        std::vector<size_t> order(upgrades.size());
        std::iota(order.begin(), order.end(), 0);
```